core/chains/evm/bulletprooftxmanager/eth_broadcaster.go

```diff
@@ -31,6 +31,21 @@ const InFlightTransactionRecheckInterval = 1 * time.Second
 
 var errEthTxRemoved = errors.New("eth_tx removed")
 
+// TransmitCheckerFactory creates a transmit checker based on a spec.
+type TransmitCheckerFactory interface {
+
+	// BuildChecker builds a new TransmitChecker based on the given spec.
+	BuildChecker(spec TransmitCheckerSpec) (TransmitChecker, error)
+}
+
+// TransmitChecker determines whether a transaction should be submitted on-chain.
+type TransmitChecker interface {
```

Comment:
I think it's worth noting here that this seems to be a "soft" check, i.e, if the check itself errors out (i.e, eth call times out) then we are expected to return `nil` from `Check` and still submit the tx on-chain right?

The reason I point this out is this is somewhat non-idiomatic Go but it seems like we're doing it for a good reason, because we don't want the failure of a check executing to block executing the tx.

If we wanted to make this a "hard" check, in the sense that this MUST pass in order for a request to be submitted, then I think the signature of `Check` will have to change to 

```golang
Check(...) (shouldSubmit bool, err error)
```

Where the caller can then do

```golang
shouldSubmit, err := checker.Check(...)
if err != nil {
  // don't submit, error with checker
}

if !shouldSubmit {
  // don't submit, and forget about this tx in the queue
}

// ... submit tx ...
```

Reply:
Yeah we want a soft check here because if there's some spurious connection issue the transaction isn't necessarily failed, whereas a simulation failure or VRF already submitted case means the transaction should never be retried. Added more info to the interface describing what the behavior here is.

Reply:
It is definitively a SOFT check. If the check fails due to network error etc we want to try and send the tx anyway.

Reply:
It may be worth including a comment to make this explicit.

Reply:
I don't hate Makram's suggestion though, since it would make it more clear.

Reply:
I did have it that way at first, returning bool and error, but landed with this because of the timeout checking. If it returns (bool, err) then each individual checker needs to do an errors.Is(err, context.Canceled) so it can return true, rather than only checking it once in eth_broadcaster. I guess a benefit is that the error logging can be moved up into eth_broadcaster..

I don't have a strong opinion if you guys think that's better, easy change to make

Reply:
I think this is fine, just need to make sure to document that this is a soft check and that the failure of a check should not hinder a transaction from going through.